File: bench/promptgreen/tasks_pending/t26/hidden_tests.mjs
// Hidden test for t26 (Zeller weekday index). Never shown to an author.
const EXPECTED = '0\n4\n5\n';

export async function run(compileFn, source) {
  const r = compileFn(source);
  if (r.ok !== true) {
    return { green: false, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, reason: 'compile/run failed' } };
  }
  const lines = r.stdout.split('\n');
  const check1 = lines[0] === '0'; // 2000-01-01: January shift rule must fire; Saturday
  const check2 = lines[1] === '4'; // 2026-07-15: no shift rule needed
  const check3 = lines[2] === '5'; // 1900-03-01: century-boundary year, non-leap
  const green = check1 && check2 && check3 && r.stdout === EXPECTED;
  return { green, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, check1, check2, check3 } };
}

File: bench/promptgreen/tasks/t26/py/hidden_tests.py
import os
import subprocess
import sys

HERE = os.path.dirname(os.path.abspath(__file__))
REF = os.path.join(HERE, "reference.py")

out = subprocess.run(
    [sys.executable, REF], capture_output=True, text=True, check=True
).stdout
lines = out.splitlines()

assert lines[0] == "0", "2000-01-01: January shift rule must fire; Saturday"
assert lines[1] == "4", "2026-07-15: no shift rule needed"
assert lines[2] == "5", "1900-03-01: century-boundary year, non-leap"

print("all checks passed")
